flash/core/registry.py

```diff
@@ -0,0 +1,189 @@
+                )
+            raise MisconfigurationException(f"Key: {key} is not in {self.__repr__()}")
+
+    def __call__(self,
+                 key: str,
+                 with_metadata: bool = False,
+                 strict: bool = True,
+                 **metadata) -> Union[Callable, Dict[str, Any], List[Dict[str, Any]], List[Callable]]:
+
+        return self.get(key, with_metadata=with_metadata, strict=strict, **metadata)
+
+    def get(self,
+            key: str,
+            with_metadata: bool = False,
+            strict: bool = True,
+            **metadata) -> Union[Callable, Dict[str, Any], List[Dict[str, Any]], List[Callable]]:
+        """
+        This function is used to gather matches from the registry:
+
+        Args:
+            key: Name of the registered function.
+            with_metadata: Whether to return associated metadata used during registration.
+            strict: Whether to return all matches if higher than 1.
+            metadata: All filtering metadata used for the registry.
+
+        """
+        matches = {_hash: name for _hash, name in self._registered_functions_mapping.items() if key == name}
+        key_in = False
+        if len(matches) > 1:
+            key_in = True
+            matches = self._filter_matches_on_metadata(matches, with_metadata=with_metadata, metadata=metadata)
+            if not strict:
+                _matches = []
+                for v in matches:
+                    match = list(v.values())[0]
+                    if with_metadata:
+                        match = match["fn"]
+                    _matches.append(match)
+                return _matches
+            if len(matches) > 1:
+                raise MisconfigurationException(
+                    f"Found {len(matches)} matches within {matches}. Add more metadata to filter them out."
+                )
+            elif len(matches) == 1:
+                matches = matches[0]
+        return self.validate_matches(key, matches, with_metadata, key_in=key_in)
+
+    def remove(self, key: str) -> None:
+        matches = {hash for hash, _key in self._registered_functions_mapping.items() if key == _key}
+        for hash in matches:
+            del self._registered_functions_mapping[hash]
+            del self._registered_functions[hash]
+
+    def _register_function(
+        self, fn: Callable, name: Optional[str] = None, override: bool = False, metadata: Dict[str, Any] = None
+    ):
+        if not isinstance(fn, FunctionType) and not isinstance(fn, partial):
+            raise MisconfigurationException("``register_function`` should be used with a function")
+
+        name = name or fn.__name__
+
+        registered_function = {"fn": fn, "name": name, "metadata": metadata}
+
+        hash_algo = hashlib.sha256()
+        hash_algo.update(str(name + str(metadata)).encode('utf-8'))
+        hash = hash_algo.hexdigest()
```

Comment:
why not just `hashkey = hash(name + str(metadata))`?

Reply:
or even better, why compute the hash at all and not just use the `name + str(metadata)` as the dict key?

Reply:
I think it looks a bit cleaner when you print to have proper hash.